File: 1885-count-number-of-homogenous-substrings/count-number-of-homogenous-substrings.cpp
class Solution {
public:
    int countHomogenous(string s) {
        //Unreal magic, SEE EDITORIAL
        //TC - O(n) and sC - O(1)
        int n = s.size();
        int ans = 1;
        int curr = 1;
        int m = 1000000007;
        for(int i = 1; i < n; i++){
            if(s[i] == s[i-1])curr++;
            else curr = 1;
            ans = (ans + (curr%m) ) % m;
        }
        return ans;
    }
};

File: 1885-count-number-of-homogenous-substrings/count-number-of-homogenous-substrings.py
class Solution:
    def countHomogenous(self, s: str) -> int:
        mod = 1000000007
        n = len(s)
        ans = 1
        curr = 1
        for i in range(1,n):
            if s[i] == s[i-1]:
                curr += 1
            else:
                curr = 1
            ans = (ans + (curr%mod))%mod
        return ans